obsolete_functions.js

```diff
@@ -0,0 +1,187 @@
+
+
+// minstall.js
+// this method isn't needed because npm4 doesn't create dead symlinks
+// and incompatible versions created through symlinks are cleaned up
+// in removeContradictingInstalledDependencies.
+// The dead symlinks created with npm5 are '--clean'ed pre-install anyway.
+function removeDeadSymlinksInInstalledDependencies() {
+  let folderPaths = [];
+  let flattenedFullFilePathArray = [];
+  let flattenedScopedFolderPaths = [];
+  let folderToCheck = [];
+
+  return moduletools.getAllModulesAndInstalledDependenciesDeep()
+    .then((result) => {
+      const localModules = result.modules;
+      folderPaths = localModules.map((localModule) => {
+        return path.join(localModule.fullModulePath, 'node_modules');
+      });
+
+      folderPaths.push('.');
+
+      return Promise.all(folderPaths.map((folderPath) => {
+        return systools.readdir(folderPath);
+      }));
+    })
+    .then((filesArray) => {
+      const fullFilePaths = filesArray.map((files, index) => {
+        return files.filter((filename) => {
+          return !filename.startsWith('@') && !filename.startsWith('.');
+        }).map((filename) => {
+          return path.join(folderPaths[index], filename);
+        });
+      });
+
+      const scopedFolderPaths = filesArray.map((files, index) => {
+        return files.filter((filename) => {
+          return filename.startsWith('@');
+        }).map((filename) => {
+          return path.join(folderPaths[index], filename);
+        });
+      });
+
+
+      // flatten the filePaths
+      flattenedFullFilePathArray = [].concat.apply([], fullFilePaths);
+      flattenedScopedFolderPaths = [].concat.apply([], scopedFolderPaths);
+
+      return Promise.all(flattenedScopedFolderPaths.map((scopedFolderPath) => {
+        return systools.readdir(scopedFolderPath);
+      }));
+    })
+    .then((scopedDependencyFolder) => {
+      const fullScopedFolderPaths = scopedDependencyFolder.map((files, index) => {
+        return files.filter((filename) => {
+          return !filename.startsWith('.');
```

Comment:
you are correct. fyi: the intend is to filter folder-names that start with `.`, because these are definitely not modules, but some npm-internal folders like `.staging`. we must not touch these